pkg/controller/model_binding_pair.go

```diff
@@ -288,6 +289,45 @@ func buildModelBindingPair(mbPair *types.ModelBindingPair) *types.ModelBindingPa
 					}
 				}
 			}
+
+			// Does this namespace contain any generic components
+			genericComponents := mbPair.Model.Spec.GenericComponents
+			for _, generic := range genericComponents {
+				_, found := compNames[generic.Name]
+				if found {
+					namespace, _ := util.GetComponentNamespace(generic.Name, mbPair.Binding)
+					genericLabels := util.GetManagedBindingLabels(mbPair.Binding, mc.Name)
+
+					// Create the k8s deployment for this generic component
+					deploy := genericcomp.NewDeployment(generic, namespace, genericLabels)
+					mc.Deployments = append(mc.Deployments, deploy)
+
+					// Create k8s service for this generic component
+					service := genericcomp.NewService(generic, namespace, genericLabels)
+					if service != nil {
+						mc.Services = append(mc.Services, service)
+					}
+
+					// Get all the secrets required by the generic component deployment
+					secrets := genericcomp.GetSecrets(*deploy)
+					for _, secret := range secrets {
+						addSecret(mc, secret, namespace)
+					}
+
+					mc.GenericComponents = append(mc.GenericComponents, &generic)
+
+					// NOTE:
+					// Initial implementation uses the first service port value for the ingress connection.  In the
```

Comment:
Has this note been captured in a Jira?

Reply:
I have a number of Jiras in enter for generic components including one for this issue.